src/sentry/api/serializers/models/group.py

```diff
@@ -23,26 +23,57 @@
 @register(Group)
 class GroupSerializer(Serializer):
     def _get_subscriptions(self, item_list, user):
-        default_subscribed = UserOption.objects.get_value(
+        results = {group.id: None for group in item_list}
+
+        # First, the easy part -- if there is a subscription record associated
+        # with the group, we can just use that to know if a user is subscribed
+        # or not.
+        subscriptions = GroupSubscription.objects.filter(
+            group__in=results.keys(),
             user=user,
-            project=None,
-            key='workflow:notifications',
         )
-        if default_subscribed == UserOptionValue.participating_only:
-            subscriptions = set(GroupSubscription.objects.filter(
-                group__in=item_list,
+
+        for subscription in subscriptions:
+            results[subscription.group_id] = subscription.is_active
+
+        # For any group that doesn't have a subscription associated with it,
+        # we'll need to fall back to the project's option value, so here we
+        # collect all of the projects to look up, and keep a set of groups that
+        # are part of that project. (Note that the common -- but not only --
+        # case here is that all groups are part of the same project.)
+        projects = defaultdict(set)
+        for group in item_list:
+            if results[group.id] is None:
+                projects[group.project].add(group.id)
+
+        if projects:
+            # This is the user's default value for any projects that don't have
+            # the option value specifically recorded. (The default "all
+            # conversations" value is convention.)
+            default = UserOption.objects.get_value(
                 user=user,
-                is_active=True,
-            ).values_list('group_id', flat=True))
-        else:
-            subscriptions = set([i.id for i in item_list]).difference(
-                GroupSubscription.objects.filter(
-                    group__in=item_list,
-                    user=user,
-                    is_active=False,
-                ).values_list('group_id', flat=True),
+                project=None,
+                key='workflow:notifications',
+                default=UserOptionValue.all_conversations,
             )
-        return subscriptions
+
+            # If you're subscribed to all notifications for the project, that
+            # means you're subscribed to all of the groups. Otherwise you're
+            # not subscribed to any of these leftover groups.
+            for project, group_ids in projects.items():
+                is_subscribed = UserOption.objects.get_value(
```

Comment:
I don't have a strong opinion here but was hesitant to circumvent the manager here (mostly for future-proofing.) The common case here is that we're only querying for one project, but if you feel strongly about this I'll change it.